src/python/grpcio/grpc/_server.py

```diff
@@ -561,6 +599,8 @@ def _send_message_callback_to_blocking_iterator_adapter(
             send_response_callback(response)
             if not _is_rpc_state_active(state):
                 break
+            with state.condition:
+                state.disable_next_compression = False
```

Comment:
Sorry I failed to catch this earlier. Setting `disable_next_compression` back to `False` here doesn't cover another branch in `_stream_response_in_pool` around line 572. I understand that is only used by an internal-experimental-hidden API, but it might one day evolve to a public API.

If possible, I think a fix now will reduce our future pain. One potential solution to this is flip the boolean value in `_state_to_send_message_op_flags` to make sure it only apply to one response message.

Reply:
I won't argue against extending compression support to the internal non-blocking API, because the health check service could perhaps benefit from compression. But I certainly cannot support making the internal non-blocking API public in its current state. As it is, there are several other features that will not work properly with it.